cohorts/dataframe_loader.py

```diff
@@ -22,14 +24,24 @@ class DataFrameLoader(object):
         The name of the dataframe, to easily reference it.
     load_dataframe : function
         A function that returns the `DataFrame` object.
-    join_on : str
+    join_on_right : str
         The column of the `DataFrame` to join on (i.e. the patient
         ID column name).
+    join_on_left: str
+        The corresponding column in the `cohorts.Patient.additional_data` 
+	to join on, if not the `id` (which is the default).
     """
     def __init__(self,
                  name,
                  load_dataframe,
-                 join_on="patient_id"):
+                 join_on=None,
+                 join_on_right="patient_id",
+                 join_on_left="patient_id"):
         self.name = name
         self.load_dataframe = load_dataframe
-        self.join_on = join_on
+        if join_on:
```

Comment:
Nit: I think `if join_on is not None` is less confusing, since `if join_on` would jump to the else if `join_on` is `""`, `False`, etc. https://stackoverflow.com/questions/7816363/if-a-vs-if-a-is-not-none

From PEP:

> Also, beware of writing if x when you really mean if x is not None -- e.g. when testing whether a variable or argument that defaults to None was set to some other value. The other value might have a type (such as a container) that could be false in a boolean context!
https://www.python.org/dev/peps/pep-0008/#id51

In this case I don't think it matters, but just a general comment :)